onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_api_helper.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h"
+
+#ifdef __ANDROID__
+#include <android/api-level.h>
+#endif
+
+struct NnApi;
+
+namespace onnxruntime {
+namespace nnapi {
+
+class ModelBuilder;
+
+enum class TargetDeviceOption : int8_t {
+  ALL_DEVICES,  // use all available target devices
+
+  /* TODO support these options
+  PREFERRED_DEVICES,  // Use one or more preferred devices (must be given)
+  EXCLUDED_DEVICES,   // Exclude one or more devices (must be given)
+   */
+
+  CPU_DISABLED,  // use all available target devices except CPU
+  CPU_ONLY,      // use CPU only
+};
+
+int32_t GetNNAPIRuntimeFeatureLevel(const ::NnApi* nnapi);
```

Comment:
nit: can this take a reference instead? the code that attempts to get the NnApi pointer should be failing at that point if it's nullptr and no other code requiring the NnApi pointer should be called.

Reply:
Sure. It can take static qualifier also.